core/shared/src/main/scala-3/zio/prelude/Validator.scala

```diff
@@ -0,0 +1,24 @@
+package zio.prelude
+
+import scala.quoted._
+
+abstract class Validator[A](
+  val assertion: Assertion[A]
+) extends Liftables {
+
+  final def validateInlineImpl(expr: Expr[A])(using Quotes, Type[A]): Expr[A] = {
+    expr match {
+      case LiteralUnlift(a) =>
```

Comment:
IMO it would be nicer to use FromExpr here, but for now I stayed consistent with the existing macro